save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ProjectSettingsMenu.kt

```diff
@@ -66,6 +73,21 @@ fun projectSettingsMenu(
     val projectPath = props.project.let { "${it.organization.name}/${it.name}" }
 
     val (wasConfirmationModalShown, setWasConfirmationModalShown) = useState(false)
+    val (isOpenGitWindow, setOpenGitWindow) = useState(false)
+    val (gitDto, setGitDto) = useState(props.gitInitDto)
+
+    child(
+        runSettingGitWindow(
```

Comment:
Also, here you make the same error with functional components, that we have made multiple times already. You create a new instance of `fc` by calling the function on every render - in the past it used to lead to lost focus after every character input. The correct way is to create an `fc` once as a top-level property and then add is as a child. If you need to pass some fields that are only available inside the parent component (`projectSettingsView`) you should add the into `props` of the child component instead of passing as function parameters.

This way the `fc` will be created only once and then only re-rendered.

Reply:
Fixed

Reply:
Not exactly this - you've simply moved component creation from `child` function several lines up. It's still created inside the render logic (i.e. all logic in `fc` is called on every render, only hooks persist between those calls). It should be created outside of `fc`, e.g. as a top level property in this file

Reply:
Yes. Changed